### PR TITLE
FI-706 Add test for Bulk Data Server DELETE request

### PR DESCRIPTION
**Pull request by yunwwang**
Add BDE-10 to test DELETE request

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR:
https://oncprojectracking.healthit.gov/support/browse/FI-706
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/modules/onc_program/bulk_data_export_sequence.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ def check_capability_statement
           break if operation.present?
         end
 
-        assert operation.present?, 'Server CapabilityStatement did not declare export operation in Group resource.'
+        assert operation.present?, 'Server CapabilityStatement did not declare support of export operation in Group resource.'
```
Comment (Jammjammjamm) [inline lib/modules/onc_program/bulk_data_export_sequence.rb]: I think these should all be `support for`.

File: lib/modules/onc_program/bulk_data_export_sequence.rb [hunk 2]
```diff
@@ -276,6 +276,29 @@ def assert_requires_access_token(status_response = @status_response)
...
+
+        assert_response_accepted(reply)
+
+        content_location = reply.headers[:content_location]
+
+        assert content_location.present?, 'Export response header did not include "Content-Location"'
+
+        reply = LoggedRestClient.delete(content_location, build_header({ accept: 'application/json' }))
+
+        assert_response_accepted(reply)
```
Comment (Jammjammjamm) [inline lib/modules/onc_program/bulk_data_export_sequence.rb]: We check that the server accepts the deletion request, but is it possible to also check that the request actually was deleted?

Comment (yunwwang) [reply lib/modules/onc_program/bulk_data_export_sequence.rb]: The problem is that IG does not specify that server shall stop the export process immediately or stop the process at all.

File: lib/modules/onc_program/bulk_data_export_sequence.rb (hunk 1 above)
Comment (yunwwang) [reply lib/modules/onc_program/bulk_data_export_sequence.rb]: OK. "support for"

File: lib/modules/onc_program/bulk_data_export_sequence.rb [hunk 3]
```diff
@@ -276,6 +276,29 @@ def assert_requires_access_token(status_response = @status_response)
         assert_requires_access_token
       end
 
+      test :bulk_data_delete_test do
+        metadata do
+          id '10'
+          name 'Bulk Data support DELETE request'
```
Comment (Jammjammjamm) [inline lib/modules/onc_program/bulk_data_export_sequence.rb]: I think this name should more closely resemble the other test names.
